cmd/mistryd/worker.go

```diff
@@ -201,40 +240,40 @@ func (s *Server) Work(ctx context.Context, j *Job) (buildInfo *types.BuildInfo,
 	}
 
 	var outErr strings.Builder
-	buildInfo.ExitCode, err = j.StartContainer(ctx, s.cfg, client, out, &outErr)
+	j.BuildInfo.ExitCode, err = j.StartContainer(ctx, s.cfg, client, out, &outErr)
 	if err != nil {
 		err = workErr("could not start docker container", err)
 		return
 	}
 
-	biJSON, err = json.Marshal(buildInfo)
+	err = out.Sync()
 	if err != nil {
-		err = workErr("could not serialize build info", err)
+		err = workErr("could not flush the output log", err)
 		return
 	}
-	err = ioutil.WriteFile(j.BuildInfoFilePath, biJSON, 0666)
+
+	finalLog, err := ReadJobLogs(j.PendingBuildPath)
 	if err != nil {
-		err = workErr("could not write build info to file", err)
+		err = workErr("could not read the job logs", err)
 		return
 	}
 
-	// fill the buildInfo Log from the freshly written log
-	err = out.Sync()
+	j.BuildInfo.Log = string(finalLog)
+	j.BuildInfo.ErrLog = outErr.String()
+	j.BuildInfo.Duration = time.Now().Sub(start).Truncate(time.Millisecond)
+
+	biJSON, err = json.Marshal(j.BuildInfo)
```

Comment:
By moving JSON marshaling after filling the `.Log` attribute, the JSON now also contains the logs. This was not the case before. I remember with did this with @apostergiou on purpose, but I have forgotten why.

Reply:
I think because the JSON was serialized to disk in a file andwe did not want the logs to be contained there

Reply:
https://github.com/skroutz/mistry/pull/65

Reply:
Makes sense. I think implementing a custom marshaler is the way here, so that we are explicit and say "we do *not* want logs to be marshaled".